### PR TITLE
[ROCm] Enabling rpc tests on ROCm environment

### PR DESCRIPTION
**Pull request by pruthvistony**

### CODE REVIEW

**Comment by pruthvistony** [comment]
@jeffdaily @sunway13 @jithunnair-amd 
Please review this change.

File: torch/testing/_internal/distributed/nn/api/remote_module_test.py [hunk 1]
```diff
@@ -17,6 +18,7 @@
 
 _PARAM_VAL = torch.nn.Parameter(torch.ones(1))
 
+TEST_WITH_ROCM = os.getenv('PYTORCH_TEST_WITH_ROCM', '0') == '1'
```
Comment (jithunnair-amd) [inline torch/testing/_internal/distributed/nn/api/remote_module_test.py:21]: Please reuse the variable as follows: `from torch.testing._internal.common_utils import TEST_WITH_ROCM`

**Review by jithunnair-amd — CHANGES_REQUESTED** [review]

File: torch/testing/_internal/distributed/nn/api/remote_module_test.py (hunk 1 above)
Comment (pruthvistony) [reply torch/testing/_internal/distributed/nn/api/remote_module_test.py]: done

**Comment by pruthvistony** [comment]
Checking on the failing test cases.

File: torch/testing/_internal/distributed/nn/api/remote_module_test.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/python3
+import os
```
Comment (rohan-varma) [inline torch/testing/_internal/distributed/nn/api/remote_module_test.py:2]: nit: unused

**Comment by pritamdamania87** [comment]
Looks like a bunch of RPC tests are failing on ROCm: https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-linux-bionic-rocm4.0.1-py3.6-test2/6414/console

**Comment by pruthvistony** [comment]
> Looks like a bunch of RPC tests are failing on ROCm: https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-linux-bionic-rocm4.0.1-py3.6-test2/6414/console

Yes, checking on the failure. I verified locally on older commit, now checking for the root cause of these new failures.

**Comment by pruthvistony** [comment]
The cases for tensorpipe GPU-to-GPU communications are failing. To enable this on ROCm, hipifying tensorpipe is required, so closing this PR for now and will later raise it after the required changes have been merged.

**Pull request closed without merging**